ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -87,18 +88,26 @@ public SchemaKStream(
     this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
     this.functionRegistry = functionRegistry;
     this.schemaRegistryClient = schemaRegistryClient;
+    this.keySerde = Objects.requireNonNull(keySerde, "keySerde");
   }
 
   public QueuedSchemaKStream toQueue() {
-    return new QueuedSchemaKStream(this);
+    return new QueuedSchemaKStream<>(this);
+  }
+
+  public Serde<K> getKeySerde() {
+    return keySerde;
+  }
+
+  public boolean isWindowed() {
+    return !(keySerde instanceof Serdes.StringSerde);
```

Comment:
perhaps we should do `return keySerde.serializer() instanceof WindowedSerializer` ?